From fe213940284d65fd72fcd0bb1d764da0d84407ec Mon Sep 17 00:00:00 2001
From: Xin Ouyang <Xin.Ouyang@windriver.com>
Date: Thu, 22 Aug 2013 13:37:23 +0800
Subject: [PATCH] policy/modules/system/selinuxutil: allow secadm to manage
 selinux config files

Fixes:
$ newrole -r secadm_r
$ cp /etc/selinux/wr-mls/setrans.conf /etc/selinux/wr-mls/setrans.conf.orig
cp: cannot create regular file '/etc/selinux/wr-mls/setrans.conf.org': Permission denied

Upstream-Status: Pending

Signed-off-by: Xin Ouyang <Xin.Ouyang@windriver.com>
Signed-off-by: Joe MacDonald <joe_macdonald@mentor.com>
Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/selinuxutil.if | 1 +
 policy/modules/system/userdomain.if  | 4 ++++
 2 files changed, 5 insertions(+)

diff --git a/policy/modules/system/selinuxutil.if b/policy/modules/system/selinuxutil.if
index 6dcf34d4e..da0c670bd 100644
--- a/policy/modules/system/selinuxutil.if
+++ b/policy/modules/system/selinuxutil.if
@@ -674,6 +674,7 @@ interface(`seutil_manage_config',`
 	')
 
 	files_search_etc($1)
+	manage_dirs_pattern($1, selinux_config_t, selinux_config_t)
 	manage_files_pattern($1, selinux_config_t, selinux_config_t)
 	read_lnk_files_pattern($1, selinux_config_t, selinux_config_t)
 ')
diff --git a/policy/modules/system/userdomain.if b/policy/modules/system/userdomain.if
index 3d166c109..93da18d63 100644
--- a/policy/modules/system/userdomain.if
+++ b/policy/modules/system/userdomain.if
@@ -1472,6 +1472,10 @@ template(`userdom_security_admin_template',`
 	logging_read_audit_config($1)
 
 	seutil_manage_bin_policy($1)
+	seutil_manage_default_contexts($1)
+	seutil_manage_file_contexts($1)
+	seutil_manage_module_store($1)
+	seutil_manage_config($1)
 	seutil_run_checkpolicy($1, $2)
 	seutil_run_loadpolicy($1, $2)
 	seutil_run_semanage($1, $2)
-- 
2.17.1

